tests/examples/components/shared/test_aml_comm.py

```diff
@@ -0,0 +1,181 @@
+import os
+import sys
+import time
+import unittest
+from unittest.mock import MagicMock
+import multiprocessing as mp
+
+import torch
+
+from utils import get_free_port
+
+sys.path.append(
+    os.path.abspath(os.path.join(os.path.dirname(__file__), "..", "..", "..", ".."))
+)
+from examples.components.shared.aml_comm import AMLCommSocket, AMLCommRedis
+from examples.components.shared.aml_smpc import AMLSMPC
+
+
+# Initialize the communication channel and afterwards test sending and receiving messages
+def init_send_recv(
+    rank,
+    world_size,
+    run_id,
+    host,
+    port,
+    msg_send,
+    msg_recv,
+    shared_dict,
+    encrypted=False,
+    use_redis=False,
+):
+    encryption = None
+    if encrypted:
+        encryption = AMLSMPC()
+
+    if rank != 0:
+        # Make sure the first process is started before the others
+        time.sleep(1)
+    if use_redis:
+        comm = TestAMLCommRedis(
+            shared_dict, rank, world_size, run_id, host, port, encryption=encryption
+        )
+    else:
+        comm = AMLCommSocket(
+            rank, world_size, run_id, host, port, encryption=encryption
+        )
+
+    if rank == 0:
+        comm.send(msg_send, 1)
+        result = comm.recv(1) == msg_recv
+    else:
+        result = comm.recv(0) == msg_recv
+        comm.send(msg_send, 0)
+
+    if hasattr(result, "__len__"):
+        shared_dict[rank] = torch.all(result)
+    else:
+        shared_dict[rank] = result
+
+
+# Mock the redis connection to avoid having to run a redis server
+class TestAMLCommRedis(AMLCommRedis):
+    def __init__(self, buffer, *args, **kwargs):
+        self.buffer = buffer
+        self._get_connection_string = MagicMock(return_value="localhost")
+        self._format_connection_string = MagicMock(return_value="localhost")
+        self._wait_for_connection = MagicMock()
+        self._connect_to_redis = MagicMock()
+        super().__init__(*args, **kwargs)
+        self._client = None
+
+    def _get_session_id(self, source, destination, flag=None):
+        session_id = super()._get_session_id(source, destination, flag)
+        if session_id not in self.buffer:
+            self.buffer[session_id] = []
+        return session_id
+
+    def _send(self, data, session_id, _) -> None:
+        self.buffer[session_id] = self.buffer[session_id] + [data]
+
+    def _recv(self, session_id, _) -> bytes:
+        while len(self.buffer[session_id]) == 0:
+            time.sleep(1)
+        item = self.buffer[session_id][0]
+        self.buffer[session_id] = self.buffer[session_id][1:]
+        return item
+
+
+class TestAMLComm(unittest.TestCase):
+    def test_aml_comm_socket_simple(self):
+        TEST_MSG = lambda recv: f"Message to {recv}"
```

Comment:
Would it be worth to also test the case of degenerate empty messages? (I guess we wouldn't encounter them in practice, but just in case, it might be worth testing those edge cases too)